src/tools/compiletest/src/common.rs

```diff
@@ -477,6 +477,24 @@ impl Config {
         ASM_SUPPORTED_ARCHS.contains(&self.target_cfg().arch.as_str())
     }
 
+    pub fn has_atomic(&self, size: &str) -> bool {
+        for config in
+            rustc_output(self, &["--print=cfg", "--target", &self.target], Default::default())
+                .trim()
+                .lines()
+        {
+            let Some((key, value)) = config.split_once("=\"").map(|(k, v)| {
+                (k, v.strip_suffix('"').expect("key-value pair should be properly quoted"))
+            }) else {
+                continue;
+            };
+            if key == "target_has_atomic" && value == size {
+                return true;
+            }
+        }
+        false
+    }
```

Comment:
Remark (for myself): I should review how we are handling rustc `--print` outputs.

Reply:
Suggestion: actually, can this just use a regex (lazily/once initialized ofc)? Like

```rs
static TARGET_HAS_ATOMIC: LazyLock<Regex> =
    LazyLock::new(||
        Regex::new(r"target_has_atomic=\"(?<width>[0-9A-Za-z]+)\"").unwrap()
    );
```